array/DNAm/preprocessing/QC.rmd

```diff
@@ -514,11 +514,10 @@ This step uses the 59 SNP probes on the array to identify genetically identical
 
 ```{r, duplicateSamples, echo = FALSE, fig.width = 8}
 
-for(i in 1:ncol(snpCor)){
-  snpCor[i,i]<-NA
-}
- hist(snpCor, breaks = seq(-1,1,0.05), xlab = "SNP correlation", main = "All samples", col = "gray")
+diag(snpCor)<-NA
+hist(snpCor, breaks = seq(-1,1,0.05), xlab = "SNP correlation", main = "All samples", col = "gray")
 abline(v = 0.8, col = "red")
+diag(snpCor)<-1

```

Comment:
Took a while for me to understand why this fixes the problem. 

This change ensures that:

```R
    # pull out samples that are genetically identical
    predictedduplicates <- vector(length = ncol(snpCor))
    for(i in seq_len(ncol(snpCor))) {
      predictedduplicates[i] <- paste(
	sort(names(which(snpCor[, i] > 0.8))),
	sep = "|",
	collapse = "|"
      )
```

still pulls out the diagonals (which we want, as a sample with no duplicates needs to be in `predictedduplicates` on its own)